packages/generate-prism-languages/index.ts

```diff
@@ -0,0 +1,106 @@
+import flowRight from "lodash.flowright"
+import pc from "picocolors"
+import { readFile, writeFile, access } from "node:fs/promises"
+import { constants } from "node:fs"
+import { join, dirname } from "node:path"
+import { languages as prismLanguages } from "prismjs/components"
+import uglify from "uglify-js"
+
+export const languagesToBundle = <const>[
+  "jsx",
+  "tsx",
+  "swift",
+  "kotlin",
+  "objectivec",
+  "rust",
+  "graphql",
+  "yaml",
+  "go",
+  "cpp",
+  "markdown",
+]
+
+/**
+ * We need to disable typechecking on this generated file as it's just concatenating JS code
+ * that starts off assuming Prism lives in global scope. We also need to provide Prism as that
+ * gets passed into an iffe preventing us from needing to use global scope.
+ */
+const header = `// eslint-disable-next-line @typescript-eslint/ban-ts-comment\n// @ts-nocheck\nimport Prism from "prismjs"\n`
+const prismPath = dirname(require.resolve("prismjs"))
+
+const readLanguageFile = async (language: string): Promise<string> => {
+  const pathToLanguage = join(prismPath, `components/prism-${language}.js`)
+  await access(pathToLanguage, constants.R_OK)
+  const buffer = await readFile(pathToLanguage, { encoding: "utf-8" })
+  return buffer.toString()
+}
+
+const strArrayFromUnknown = (input: unknown) => (array: string[]) => {
+  if (typeof input === "string") array.push(input)
+  else if (Array.isArray(input)) {
+    array = [...array, ...input.filter(i => typeof i === "string")]
```

Comment:
Can we just array.concat here?

Reply:
Does that remove falsey values? I noticed there are times where a peer dep can be an array of nulls. I could also run `.concat(...input).filter(Boolean)`

Reply:
Oh I was thinking .concat(input.filter()), but could also do concat(input).filter() as well!